downstairs/src/extent_inner_sqlite.rs

```diff
@@ -75,33 +75,63 @@ impl ExtentInner for SqliteInner {
         new_gen: u64,
         job_id: JobOrReconciliationId,
     ) -> Result<(), CrucibleError> {
+        if !self.dirty()? {
+            /*
+             * If we have made no writes to this extent since the last flush,
+             * we do not need to update the extent on disk
+             */
+            return Ok(());
+        }
+
+        // We put all of our metadata updates into a single write to make this
+        // operation atomic.
+        self.set_flush_number(new_flush, new_gen)?;
```

Comment:
I'm suspicious about this change.

If we lose power right after this line, the metadata DB will tell us that we're at a particular flush (`new_flush`) and `dirty = 0`, but the data in the extent file has not necessarily been persisted to disk!

I think the existing implementation – which updates flush number in the DB **after** syncing the extent file – is correct.